src/xtgeo/surface/_regsurf_export.py

```diff
@@ -167,60 +130,28 @@ def _export_irap_binary_python(self, mfile):
     if (inum % nchunk) > 0:
         chunks.append(inum % nchunk)
     start = 0
+    data = bytearray(header)
+    chunk_size = nchunk * 4
+
+    # Precompute the struct.pack format for chunk size
+    chunk_size_pack = struct.pack(">i", chunk_size)
+
     for chunk in chunks:
-        ap += struct.pack(">i", chunk * 4)
-        ap += struct.pack(f">{chunk:d}f", *vals[start : start + chunk])
-        ap += struct.pack(">i", chunk * 4)
+        chunk_data = np.array(vals[start : start + chunk], dtype=">f4").tobytes()
+        data.extend(chunk_size_pack)
+        data.extend(chunk_data)
+        data.extend(chunk_size_pack)
         start += chunk
 
     if mfile.memstream:
-        mfile.file.write(ap)
+        mfile.file.write(data)
     else:
         with open(mfile.name, "wb") as fout:
-            fout.write(ap)
-
-
-def _export_irap_binary_cxtgeo(self, mfile):
-    """Export to Irap binary using C backend.
-
-    Args:
-        mfile (FileWrapper): xtgeo file instance
-
-    Raises:
-        RuntimeError: Export to Irap Binary went wrong...
-    """
-    vals = self.get_values1d(fill_value=UNDEF_MAP_IRAPB, order="F")
-    ier = _cxtgeo.surf_export_irap_bin(
-        mfile.get_cfhandle(),
-        self._ncol,
-        self._nrow,
-        self._xori,
-        self._yori,
-        self._xinc,
-        self._yflip * self._yinc,
-        self._rotation,
-        vals.astype(np.float64),
-        0,
-    )
-
-    if ier != 0:
-        mfile.cfclose(strict=False)  # strict False as C routine may have closed
-        raise RuntimeError(f"Export to Irap Binary went wrong, code is {ier}")
-
-    mfile.cfclose()
+            fout.write(data)
 
 
 def export_ijxyz_ascii(self: RegularSurface, mfile: FileWrapper) -> None:
-    # cxtgeo is default since twice as fas as python, but use python if memstreams
-    if mfile.memstream:
-        _export_ijxyz_ascii_python(self, mfile)
-    else:
-        _export_ijxyz_ascii_cxtgeo(self, mfile)
-
-
-def _export_ijxyz_ascii_python(self: RegularSurface, mfile: FileWrapper) -> None:
-    """Export to DSG IJXYZ ascii format, using python."""
-    # the python version is ~twice as slow as the cxtgeo version
+    """Export to DSG IJXYZ ascii format."""
```

Comment:
Included this also, although it is not Irap format, it was a low hanging fruit having both C and python support initially. So simply removed the C version.